File: 9/1.js
const { sum } = require('../common.js')
const { calculateNextValue } = require('./common.js')

const input = require('node:fs').readFileSync('input', { encoding: 'latin1' });
const lines = input.trim().split('\n').map(line => line.split(' ').map(Number));

console.log(sum(lines.map(sequence => calculateNextValue(sequence))))

File: 18/1.py
from common import Point, directions, move, total

with open('input', encoding = 'latin1') as input:
	lines = [line.strip() for line in input]

position = Point(0, 0)
nodes = [position]
perimeter = 0

for line in lines:
	fields = line.split()[:2]
	direction = directions[fields[0]]
	distance = int(fields[1])

	perimeter += distance
	position = move(position, direction, distance)
	nodes.append(position)

print(total(nodes, perimeter))
